tests/4.5/application_kernels/qmcpack_target_math.c

```diff
@@ -0,0 +1,46 @@
+//===--- qmcpack_target_math.c --- math lib invocation inside target---------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+//
+////===----------------------------------------------------------------------===//
+
+#include <assert.h>
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include "ompvv.h"
+
+#define N 1000
+
+int test_math_lib_inside_target() {
+  OMPVV_INFOMSG("test_math_lib_inside_target");
+
+  double array[N];
+  int errors = 0;
+  
+  // Array initialization
+  for (int i = 0; i < N; ++i) {
+    array[i] = 0.99;
+  }
+
```

Comment:
Let us make this test closer to the real code and more challenging.
Please a dummy line
`int c99_zero = FP_ZERO;`
This should choke clang in the CUDA pass.

Reply:
FP_ZERO is for floating point. Is there some reason you want to assign it to int ?

Reply:
FP_ZERO is one of the return value of fpclassify, the return type is int.
https://www.gnu.org/software/libc/manual/html_node/Floating-Point-Classes.html